app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesFragment.kt

```diff
@@ -0,0 +1,511 @@
+                }
+            }
+        }.flow.cachedIn(viewModelScope)
+    }.collectAsLazyPagingItems()
+
+    val refreshState = rememberPullToRefreshState {
+        data.loadState.refresh is LoadState.NotLoading
+    }
+
+    if (refreshState.isRefreshing) {
+        LaunchedEffect(Unit) {
+            data.refresh()
+        }
+    }
+
+    val searchBarConnection = remember {
+        object : NestedScrollConnection {
+            override fun onPostScroll(consumed: Offset, available: Offset, source: NestedScrollSource): Offset {
+                searchBarOffsetY = (searchBarOffsetY + consumed.y).roundToInt().coerceIn(-300, 0)
+                return Offset.Zero // We never consume it
+            }
+        }
+    }
+    val combinedModifier = Modifier.nestedScroll(searchBarConnection).nestedScroll(refreshState.nestedScrollConnection)
+
+    var expanded by remember { mutableStateOf(false) }
+    var hidden by remember { mutableStateOf(false) }
+    var selectMode by remember { mutableStateOf(false) }
+    val checkedInfoMap = remember { mutableStateMapOf<Long, BaseGalleryInfo>() }
+    SideEffect {
+        if (checkedInfoMap.isEmpty()) selectMode = false
+        if (selectMode) expanded = true
+    }
+
+    SearchBarScreen(
+        title = title,
+        searchFieldState = searchFieldState,
+        searchFieldHint = searchBarHint,
+        onApplySearch = { switchFav(urlBuilder.favCat, it) },
+        onSearchExpanded = { hidden = true },
+        onSearchHidden = { hidden = false },
+        refreshState = refreshState,
+        searchBarOffsetY = searchBarOffsetY,
+        trailingIcon = {
+            IconButton(onClick = { activity.openSideSheet() }) {
+                Icon(imageVector = Icons.Outlined.FolderSpecial, contentDescription = null)
+            }
+        },
+    ) {
+        val realPadding = PaddingValues(
+            top = it.calculateTopPadding() + 8.dp,
+            bottom = it.calculateBottomPadding(),
+        )
+        val listMode by remember {
+            Settings.listModeBackField.valueFlow()
+        }.collectAsState(Settings.listMode)
+        val marginH = dimensionResource(id = R.dimen.gallery_list_margin_h)
+        Box(
+            modifier = Modifier.fillMaxSize(),
+            contentAlignment = Alignment.Center,
+        ) {
+            when (val state = data.loadState.refresh) {
+                is LoadState.Loading -> if (data.itemCount == 0) CircularProgressIndicator()
+                is LoadState.Error -> {
+                    Column(
+                        modifier = Modifier.widthIn(max = 228.dp).clickable { data.retry() },
+                        horizontalAlignment = Alignment.CenterHorizontally,
+                    ) {
+                        Image(
+                            imageVector = EhIcons.Big.Default.SadAndroid,
+                            contentDescription = null,
+                            modifier = Modifier.padding(16.dp),
+                        )
```

Comment:
The color and the ripple are a bit strange.

Reply:
Done